### PR TITLE
per-bucket random replica order #2957

### PR DESCRIPTION
**Pull request by yaojingguo**
@tschottdorf. After I did `git fetch origin && git reset --soft origin/master && git commit`, I am unable to push to the old branch `issue-2957`. So this new pull request.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3137)

### CODE REVIEW

**Comment by tschottdorf** [comment]
All you needed was a force push. Unfortunately, you've deleted the old branch and this one is a botched rebase, so I have no way of conveniently obtaining your previous work. Please restore your old code and force-push that to this PR's branch so we can merge it. Thanks!

**Comment by yaojingguo** [comment]
@tschottdorf , sorry for the trouble. I have created a new commit according to your suggestions.

File: kv/replica_slice.go [hunk 1]
```diff
@@ -126,11 +129,20 @@ func (rs replicaSlice) SortByCommonAttributePrefix(attrs []string) int {
 // of the slice, keeping the order of the remaining elements stable.
 // The function will panic when invoked with an invalid index.
 func (rs replicaSlice) MoveToFront(i int) {
-	if i >= len(rs) {
```
Comment (tamird) [inline kv/replica_slice.go]: these changes look spurious

**Comment by yaojingguo** [comment]
@tamird, that is my mistake. A new commit after fixing it.

**Comment by tschottdorf** [comment]
Thank you @yaojingguo!

**Pull request merged**